src/iter.php

```diff
@@ -893,6 +893,36 @@ function join(string $separator, iterable $iterable): string {
     return $str;
 }
 
+/**
+ * Splits a string by a separator
+ *
+ * Examples:
+ *
+ *      iter\split(', ', 'a, b, c')
+ *      => iterable with values 'a', 'b' and 'c'
+ *
+ * @param string $separator Separator to use between elements
+ * @param string $data The iterable to join
+ *
+ * @return iterable
+ */
+function split(string $separator, string $data): iterable
+{
+    if (strlen($separator) === 0) {
+        throw new \InvalidArgumentException('Separator must be non-empty string');
+    }
+
+    $offset = 0;
+    while (
+        $offset < strlen($data)
```

Comment:
```suggestion
        $offset < \strlen($data)
```

Same for the `strlen` use below.